doc/design/test-dandisets-1.md

```diff
@@ -0,0 +1,105 @@
+dandisets" listings (or in all listings shown to the owners). Admins would get a
+checkmark to toggle the display of test dandisets, to help simulate what other
+users are seeing when they visit the archive.
+
+To encourage the use of this feature, the dandiset creation screen would have a
+checkbox along with a description of what this is for ("If you are new to the
+archive and are just exploring, please consider using the staging server
+instead. Otherwise, select this option to avoid this dandiset from appearing in
+public searches.", with "staging server" linked thereto).
+
+### Hide draft-only dandisets
+
+Similarly to hiding explicitly marked test dandisets, we could, by default, also
+hide dandisets that have no published versions (as above, these dandisets would
+still appear for the owners). As the public may *want* to see draft-only
+dandisets, everyone would be able to toggle this mode with a setting.
+
+### Hide dandisets with no assets
+
+Another variation on the previous proposal: hide dandisets that contain no
+assets. These dandisets are a subset of draft-only dandisets, since publishing a
+dandiset requires assets to be present.
+
+This option is problematic on its own--it is a valid use case to declare
+scientific intentions by creating a draft dandiset with no assets, then collect
+and add data only after months or years of real scientific work. That is to say,
+an assetless dandiset that nonetheless has a description, funding agent,
+contact person, owners, etc. may indeed be fully valid.
+
+### Add questions to dandiset creation page
+
+Adding questions such as:
+- Do you welcome collaborators? (Y/N)
+- When do you anticipate posting data to the dandiset?
+
+may help to stem the tide of abandoned dandisets.
+
+The collaborators question, answered positively, would add a badge to the
+dandiset's landing page and search result display indicating the call for
+collaboration.
+
+The anticipation date question would delay considering the dandiset abandoned
+until after that date.
+
+### Enable deletion of dandisets
+
+Part of the problem may be that people simply cannot delete their own abanonded
+dandisets. Offering a delete button with the proper semantics could help here.
+
+The archive does not allow for deletion of *published* dandisets, by design. But
+dandisets that are draft-only should be deleteable.
+
+And, as en extension to that idea, a published dandiset should be able to be
+"sealed" by having its draft dandiset deleted, thus closing it off to further
+updates. This would serve as a signal that the dandiset is in its final state
+and has become a permanent member of the archive. Owners and admins will be able
+to "unseal" sealed dandisets in case further modification is needed.
+
+### Delisting of inactive dandisets
+
+Dandisets in a certain set of conditions will be automatically "delisted",
+meaning they will not appear in listings or searches other than to the owners
```

Comment:
listed/delisted is yet another flag to implement/manage.  Why not just to remove?

Also there is a notion of "revert 'draft' changes" which we might introduce, similarly to OpenNeuro.
FWIW, here is a recently announced OpenNeuro data retention policy sent out as email:



Dear OpenNeuro user,
We are writing to inform you that OpenNeuro's data retention policy has gone into effect as of July 18, 2022. This policy affects datasets that are in draft status, which are datasets that have had no snapshots made, or have had changes made to them since the most recent snapshot. Public and private datasets may be in draft status, and making a snapshot does not automatically make a dataset public.Starting on August 15, 2022, if a dataset has been in draft status for at least 28 days, then it MAY be reverted to the latest snapshot at any time. If no snapshot is found, this means that the dataset will be deleted.This policy is being set for two reasons: 1) Dataset owners may make changes to their datasets and unknowingly fail to snapshot these changes, making them inaccessible to other users. 2) Unsaved changes to datasets impose a higher maintenance burden on OpenNeuro, limiting our ability to make changes that can benefit all users of the site.To determine which of your datasets may be affected, you may find a full list at https://openneuro.org/search?mydatasets. On each dataset, navigate to the "Draft" version and locate the banner at the top. If the banner is green, no further action is needed. If the banner is not green, please make sure to snapshot your dataset prior to August 15, 2022. If you are reluctant to snapshot because the dataset is not in a "final" state, it is possible to release a new version and mark previous versions as "deprecated", indicating that users should update to the latest version.We will be sending additional emails to all dataset owners with affected datasets.If you have any questions regarding this policy, please feel free to submit a support ticket. Thank you,The OpenNeuro team | Dear OpenNeuro user,We are writing to inform you that OpenNeuro's data retention policy has gone into effect as of July 18, 2022. This policy affects datasets that are in draft status, which are datasets that have had no snapshots made, or have had changes made to them since the most recent snapshot. Public and private datasets may be in draft status, and making a snapshot does not automatically make a dataset public.Starting on August 15, 2022, if a dataset has been in draft status for at least 28 days, then it MAY be reverted to the latest snapshot at any time. If no snapshot is found, this means that the dataset will be deleted.This policy is being set for two reasons: 1) Dataset owners may make changes to their datasets and unknowingly fail to snapshot these changes, making them inaccessible to other users. 2) Unsaved changes to datasets impose a higher maintenance burden on OpenNeuro, limiting our ability to make changes that can benefit all users of the site.To determine which of your datasets may be affected, you may find a full list at https://openneuro.org/search?mydatasets. On each dataset, navigate to the "Draft" version and locate the banner at the top. If the banner is green, no further action is needed. If the banner is not green, please make sure to snapshot your dataset prior to August 15, 2022. If you are reluctant to snapshot because the dataset is not in a "final" state, it is possible to release a new version and mark previous versions as "deprecated", indicating that users should update to the latest version.We will be sending additional emails to all dataset owners with affected datasets.If you have any questions regarding this policy, please feel free to submit a support ticket. Thank you,The OpenNeuro team

Reply:
> listed/delisted is yet another flag to implement/manage. Why not just to remove?

This is a great point, but it represents a more aggressive stance on usage of the archive. If that's where we want to go, then great: it indeed makes developing functionality and performing maintenance much easier, but at a cost to the user's experience of using the archive. My background sense is that it's already hard enough to convince people to put their data in a public archive without also subjecting them to strict deletion policies for dandisets that have, e.g., not seen activity in a while. The problem here is (as usual for us lately) one of scale: users that have uploaded very large data such as Zarrs of sufficient size will lose all trust in the archive if we do things like delete after a period of inactivity.

All that being said, I would like to establish what our intended "terms of use" really are. That will help me tighten up the language in this design doc, and enable us to write proper documentation for users of the archive so they know what it's meant for and what to expect from it.

Reply:
The other idea here is to delist with a notification to the owner, and *then* delete after another grace period. The delisting then is simply a way to keep the archive "clean" without also permanently destroying possibly valuable data.